src/conn/conn_compact.c

```diff
@@ -258,18 +258,21 @@ __background_compact_should_skip(WT_SESSION_IMPL *session, const char *uri, int6
     WT_PREFIX_SKIP(filename, "file:");
     ret = __wt_block_manager_named_size(session, filename, &file_size);
 
-    /* It is possible that the file no longer exists. */
+    /* Ignore the error if the file no longer exists or in case of permission issues. */
     if (ret == ENOENT) {
         *skipp = true;
         return (0);
-    }
-
-    WT_RET(ret);
-
-    if (file_size <= WT_MEGABYTE) {
-        WT_STAT_CONN_INCR(session, background_compact_skipped);
-        *skipp = true;
-        return (0);
+    } else if (ret == EACCES) {
```

Comment:
Why we are not skipping this file in this case?

Reply:
I think you are right that we should skip - I am just not sure if we get permission issues for the wrong reasons when using that specific function?
What do you think @Sean04 ?

Reply:
Yes, I think we should skip. Background compact's purpose is to compact eligible files. Files we don't have access to are not eligible and should be skipped.